### DISCUSSION TITLE
Detection Threshold Lower Limit

### DISCUSSION THREAD

**Discussion by qubithaze** [blakeblackshear/frigate#12102] [General]
Is there a reason for what appears to be a globally applied detection threshold of .4 here:  https://github.com/blakeblackshear/frigate/blob/dev/frigate/object_detection.py#L222

I understand that is a default for the function parameter, but it doesn't appear to be passed:  https://github.com/blakeblackshear/frigate/blob/dev/frigate/video.py#L471

I have some lower confidence detections at night I would like to still consider valid.

**Comment by NickM-27** [comment]
there is a limit to the number of detections that can be handled internally for each detector output. In testing detections < 0.4 are not valid and often are just partial duplicates of other detections with higher scores

**Reply by qubithaze** [reply]
I'm using a custom model and in my scenario, I'd like to avoid short-circuiting the pipeline.  Any opposition to a PR that would do something to make this configurable?

![f83e68d1-8d0a-4794-a4da-69f28d069b13](https://github.com/blakeblackshear/frigate/assets/5642320/48b9da62-c8c2-4dec-a9e7-a83a593bc792)

**Reply by NickM-27** [reply]
I don't think we would want this to be configurable, we already have two fields that act similarly and there is confusion for some users, I think an additional field would just confuse things further. 

CC @blakeblackshear

**Reply by qubithaze** [reply]
I hear 'ya about causing confusion.  I can work around it by artificially boosting the confidence in my model outputs, or run a fork, so it's not a blocker.  But it did take me quite a while to figure out why my detections where being silently dropped when I had configured object filter thresholds lower than .4

**Reply by qubithaze** [reply]
It would also be nice to be able to bypass NMS for model outputs where it's already been applied.  I see it eliminating some of these detections as well when there is only a single box.

**Reply by qubithaze** [reply]
I should have started off this discussion with this comment:

Thank you guys so much for creating and maintaining this project.  It's awesome!
